internal/api/v1beta1/app_types.go

```diff
@@ -500,16 +517,47 @@ func (app *App) DoCanary(now metav1.Time) error {
 	}
 
 	if app.Spec.Canary.NextScheduledTime.Equal(&now) || app.Spec.Canary.NextScheduledTime.Before(&now) {
+
 		// update traffic weight distributions across deployments
 		app.Spec.Deployments[0].RoutingSettings.Weight = app.Spec.Deployments[0].RoutingSettings.Weight - app.Spec.Canary.StepWeight
 		app.Spec.Deployments[1].RoutingSettings.Weight = app.Spec.Deployments[1].RoutingSettings.Weight + app.Spec.Canary.StepWeight
 		app.Spec.Canary.CurrentStep++
 
+		if app.Spec.Canary.Target != nil {
+			// scale units based on weight and process target
+			for processName, target := range app.Spec.Canary.Target {
+				p1Units, p2Units := getUpdatedUnits(app.Spec.Deployments[0].RoutingSettings.Weight, target)
+				// might be fine to ignore these errors
+				if err := app.Spec.Deployments[0].setUnits(processName, p1Units); err != nil {
+					log.Printf("the process: %s is not present in the previous deployment\n", processName)
+				}
+				if err := app.Spec.Deployments[1].setUnits(processName, p2Units); err != nil {
+					log.Printf("the process: %s in not present in the updated deployment\n", processName)
```

Comment:
Does this `log.Printf` statement work well with https://github.com/shipa-corp/ketch/blob/main/internal/controllers/app_controller.go#L46?
It might be that we should have 
```golang
func (app *App) DoCanary(log logr.Logger, now metav1.Time) {
     log.Info()
}
```

Reply:
Done.